kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/balance/presenter/BalancePresenter.java

```diff
@@ -36,22 +90,149 @@ private void updateBalance(Balance balance) {
 		if (balanceValue == 0) {
 			balanceString = BALANCE_ZERO_TEXT;
 		} else {
-			balanceString = StringUtil.getAmountFormatted(balanceValue);
+			balanceString = getAmountFormatted(balanceValue);
 		}
 		if (view != null) {
 			view.updateBalance(balanceString);
 		}
 	}
 
+	private int getType(OfferType offerType) {
+		switch (offerType) {
+			case SPEND:
+				return SPEND;
+			default:
+			case EARN:
+				return EARN;
+		}
+	}
+
+	private void createOrderObserver() {
+		orderObserver = new Observer<Order>() {
+			@Override
+			public void onChanged(Order order) {
+				if (order.getOrigin() == Origin.MARKETPLACE) {
```

Comment:
why only marketplace?
Do we have a spec for message if transaction fails?

Reply:
we do not show for external
yes

Reply:
@OrenZak what happen if transaction fails should we update the subtitle (ask Noa)

Reply:
it's updated according to the status